### DISCUSSION TITLE
Securing DB user/password

### DISCUSSION THREAD

**Discussion by FX-Liagre** [keycloak/keycloak#10595] [Archive]
On the Wildfly version, we were using initially picketbox SecureIdentityLoginModule to encrypt user/pwd in standalone.xml files.
We then switched to Elytron encrypted expressions on the latest releases of Keycloak.
But I cannot find anything in the Keycloak.X docs about such possibility.
And our customers' security guys (and DB admins...) tend to faint when you present them a config text file with clear text DB credentials... :-(

Is there anything I didn't found, or?...

**Comment by pglavica** [comment]
Hi,

I also have the same issue.
Is there any INFO related to the matter?

Regards,
Patrik

**Comment by dfs-** [comment]
The only way currently is to use environment variables for the Keycloak process. 
However this isn't really _secure_ either, as the information can easily be read using tools (i.e. Process Explorer on Windows).

The same also applies to other secrets like the https-key-store-password (when using a pfx/p12).

**Reply by dfs-** [reply]
I created an issue ticket: https://github.com/keycloak/keycloak/issues/11089
